docs/user-guides/acme-dns-validation.md

```diff
@@ -0,0 +1,94 @@
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    # ACME dns-01 provider configurations
+    dns01:
+      # Here we define a list of DNS-01 providers that can solve DNS challenges
+      providers:
+      - name: prod-dns
+        clouddns:
+          # A secretKeyRef to a google cloud json service account
+          serviceAccountSecretRef:
+            name: clouddns-service-account
+            key: service-account.json
+          # The project in which to update the DNS zone
+          project: gcloud-prod-project
+      - name: cf-dns
+        cloudflare:
+          email: user@example.com
+          # A secretKeyRef to a cloudflare api key
+          apiKeySecretRef:
+            name: cloudflare-api-key
+            key: api-key.txt
+```
+
+We have specified the ACME server URL for Let's Encrypt's [staging environment][3]. The staging environment will not issue trusted certificates but is used to ensure that the verification process is working properly before moving to production. Let's Encrypt's production environment imposes much stricter [rate limits][4], so to reduce the chance of you hitting those limits it is highly recommended to start by using the staging environment. To move to production, simply change the URL to `https://acme-v01.api.letsencrypt.org/directory`.
+
+The first stage of ACME is for the client to register with the ACME server. This phase includes generating an asymmetric key pair which is then associated with the email address specified in the `Issuer`. Depending on the CA, this could be used to send expiry notices when your certificates are coming up for renewal. The generated private key is stored in a `Secret` called `letsencrypt-staging`.
+
+The `dns01` stanza contains a list of `dns-01` providers that can be used to solve DNS challenges. A list of currently supported providers can be found [here][5]
+
+Once we have created the above `Issuer` we can use it to obtain a certificate.
+
+```yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: nginx-k8s-io
+  namespace: default
+spec:
+  secretName: nginx-k8s-io-tls
+  issuerRef:
+    name: letsencrypt-staging
+  commonName: nginx.k8s.io
+  dnsNames:
+  - nginx2.k8s.io
+  - nginx3.k8s.io
+  acme:
+    config:
+    - dns01:
+        provider: clouddns
+      domains:
+      - nginx.k8s.io
+    - dns01:
+        provider: cloudflare
+      domains:
+      - nginx2.k8s.io
+      - nginx3.k8s.io
+```
+
+The `Certificate` resource describes our desired certificate and the possible methods that can be used to obtain it. You can learn more about the `Certificate` resource [here][7]. If the certificate is obtained successfully the resulting key pair will be stored in a secret called `nginx-k8s-io-tls` in the same namespace as the `Certificate`. The certificate will have a common name of `nginx.k8s.io` and the [Subject Alternative Names][6] (SANs) will be `nginx.k8s.io`, `nginx2.k8s.io` and `nginx3.k8s.io`.
```

Comment:
`If the certificate is obtained successfully` - how can I know if the certificate is obtained successfully? `kubectl describe certificate ...` (and also show some example output that the user should expect)